File: src/main/java/net/igneo/icv/networking/packet/AnimatedSyncC2SPacket.java
package net.igneo.icv.networking.packet;

import net.igneo.icv.enchantmentActions.PlayerEnchantmentActionsProvider;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.server.level.ServerLevel;
import net.minecraft.server.level.ServerPlayer;
import net.minecraftforge.network.NetworkEvent;

import java.util.function.Supplier;

public class AnimatedSyncC2SPacket {
    private final boolean state;
    
    public AnimatedSyncC2SPacket(boolean state) {
        this.state = state;
    }
    
    public AnimatedSyncC2SPacket(FriendlyByteBuf buf) {
        state = buf.readBoolean();
    }
    
    public void toBytes(FriendlyByteBuf buf) {
        buf.writeBoolean(state);
    }
    
    public boolean handle(Supplier<NetworkEvent.Context> supplier) {
        NetworkEvent.Context context = supplier.get();
        context.enqueueWork(() -> {
            ServerPlayer player = context.getSender();
            ServerLevel level = player.serverLevel();
            player.getCapability(PlayerEnchantmentActionsProvider.PLAYER_ENCHANTMENT_ACTIONS).ifPresent(enchVar -> {
                enchVar.animated = state;
            });
        });
        return true;
    }
}

File: src/main/java/net/igneo/icv/networking/packet/AnimatedSyncC2SPacket.kt
package net.igneo.icv.networking.packet

import net.igneo.icv.enchantmentActions.PlayerEnchantmentActions
import net.igneo.icv.enchantmentActions.PlayerEnchantmentActionsProvider
import net.minecraft.network.FriendlyByteBuf
import net.minecraftforge.network.NetworkEvent
import java.util.function.Supplier

class AnimatedSyncC2SPacket {
    private val state: Boolean

    constructor(state: Boolean) {
        this.state = state
    }

    constructor(buf: FriendlyByteBuf) {
        state = buf.readBoolean()
    }

    fun toBytes(buf: FriendlyByteBuf) {
        buf.writeBoolean(state)
    }

    fun handle(supplier: Supplier<NetworkEvent.Context>): Boolean {
        val context = supplier.get()
        context.enqueueWork {
            val player = context.sender
            player!!.getCapability(PlayerEnchantmentActionsProvider.PLAYER_ENCHANTMENT_ACTIONS)
                .ifPresent { enchVar: PlayerEnchantmentActions -> enchVar.animated = state }
        }
        return true
    }
}